components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -42,14 +41,15 @@ type sinkValidator func(ctx context.Context, r *Reconciler, subscription *eventi
 
 // Reconciler reconciles a Subscription object
 type Reconciler struct {
-	ctx context.Context
 	client.Client
-	Backend          handlers.MessagingBackend
+	sinkValidator
```

Comment:
You're right. I thought this might be needed for the backend you are working on. But it is required for the reconciler not the backend. I'll do it in a separate PR or create an issue for that.